File: Medium/2241. Design an ATM Machine.cpp
class ATM {
public:
    long bills[5] = {0, 0, 0, 0, 0};
    long amounts[5] = {20, 50, 100, 200, 500};

    ATM() {}

    void deposit(vector<int> banknotesCount) {
        for(int i = 0; i < 5; i++){
            bills[i] += banknotesCount[i];
        }
    }
    
    vector<int> withdraw(int amount) {
        vector<int> output = {0, 0, 0, 0, 0};

        for(int i = 4; i > -1; i--){
            long can = amount / amounts[i];
            long have = bills[i];
            long taking = have;

            if(can < have){
                taking = can;
            } 

            amount -= (taking * amounts[i]);
            bills[i] -= taking;
            output[i] += taking;
        }

        if(amount != 0){
            for(int i = 4; i > -1; i--){
                bills[i] += output.back();
                output.pop_back();
            }
            output.push_back(-1);
        }

        return output;
    }
};

/**
 * Your ATM object will be instantiated and called as such:
 * ATM* obj = new ATM();
 * obj->deposit(banknotesCount);
 * vector<int> param_2 = obj->withdraw(amount);
 */

File: Medium/2241. Design an ATM Machine.py
class ATM:

    def __init__(self):
        self.bills = [0, 0, 0, 0, 0]
        self.amounts = [20, 50, 100, 200, 500]

    def deposit(self, banknotesCount: List[int]) -> None:
        for i, x in enumerate(banknotesCount):
            self.bills[i] += x

    def withdraw(self, amount: int) -> List[int]:
        output = [0]*5

        for i in range(4,-1,-1):
            can = int(amount / self.amounts[i])
            have = self.bills[i]
            taking = have if can >= have else can

            amount -= (taking*self.amounts[i])
            self.bills[i] -= taking
            output[i] += taking

        if amount != 0:
            for i in range(5): self.bills[i] += output[i]
            return [-1]
        return output
        


# Your ATM object will be instantiated and called as such:
# obj = ATM()
# obj.deposit(banknotesCount)
# param_2 = obj.withdraw(amount)